external-crates/move/crates/move-compiler/src/linters/combinable_bool_conditions.rs

```diff
@@ -0,0 +1,157 @@
+                                    "This is always contradictory and can be simplified to false",
+                                );
+                            } else {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `>=`.",
+                                );
+                            }
+                        }
+                        (BinOp_::Ge, BinOp_::Eq) | (BinOp_::Eq, BinOp_::Ge) => {
+                            if op.value == BinOp_::And {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `==`.",
+                                );
+                            } else {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `>=`.",
+                                );
+                            }
+                        }
+                        (BinOp_::Le, BinOp_::Eq) | (BinOp_::Eq, BinOp_::Le) => {
+                            if op.value == BinOp_::And {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `==`.",
+                                );
+                            } else {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `<=`.",
+                                );
+                            }
+                        }
+                        (BinOp_::Neq, BinOp_::Lt) | (BinOp_::Lt, BinOp_::Neq) => {
+                            if op.value == BinOp_::And {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `<`.",
+                                );
+                            }
+                        }
+                        (BinOp_::Neq, BinOp_::Gt) | (BinOp_::Gt, BinOp_::Neq) => {
+                            if op.value == BinOp_::And {
+                                add_replaceable_comparison_diag(
+                                    self.env,
+                                    exp.exp.loc,
+                                    "Consider simplifying to `>`.",
+                                );
+                            }
+                        }
```

Comment:
These cases feel a bit different. It is not really "simplifying" rather the `!=` is already covered by the GT and LT

Reply:
refactored